cryptographic_estimators/MQEstimator/MQAlgorithms/dinur1.py

```diff
@@ -25,30 +25,26 @@
 
 
 class DinurFirst(MQAlgorithm):
-    r"""
-    Construct an instance of Dinur's first estimator
-
-    The Dinur's first is a probabilistic algorithm to solve the MQ problem over GF(2) [Din21a]_. It computes the parity
-    of the number of solutions of many quadratic polynomial systems. These systems come from the specialization, in the
-    original system, of the values in a fixed set of variables.
-
-
-    INPUT:
+    def __init__(self, problem: MQProblem, **kwargs):
+        """Construct an instance of Dinur's first estimator.
 
-    - ``problem`` -- MQProblem object including all necessary parameters
-    - ``nsolutions`` -- number of solutions (default: 1)
-    - ``h`` -- external hybridization parameter (default: 0)
+        The Dinur's first is a probabilistic algorithm to solve the MQ problem over GF(2) [Din21a]_. It computes the parity
+        of the number of solutions of many quadratic polynomial systems. These systems come from the specialization, in the
+        original system, of the values in a fixed set of variables.
 
-    EXAMPLES::
+        Args:
+            problem (MQProblem): MQProblem object including all necessary parameters.
+            nsolutions (int, optional): Number of solutions (default: 1).
```

Comment:
why is here an `optional` in the parameter description? Is it necessary, as its now used in the previous algorithms (`crossbreed` for example, where `max_D` is also just an optional parameter) ?

Reply:
I removed the optional mark for now.